lib/active_model_serializers/adapter/json_api/resource_identifier.rb

```diff
@@ -20,11 +20,13 @@ def as_json
 
         def type_for(serializer)
           return serializer._type if serializer._type
-          if ActiveModelSerializers.config.jsonapi_resource_type == :singular
-            serializer.object.class.model_name.singular
-          else
-            serializer.object.class.model_name.plural
+          type = serializer.object.class.to_s.split('::')
+          type.map! { |t| t.underscore.downcase }
```

Comment:
I think this should re-use or add logic in [KeyTransform](https://github.com/rails-api/active_model_serializers/blob/61412d80cabc3900dfbd886bc3f25fb0e0d39433/lib/active_model_serializers/key_transform.rb#L30-L32)

(And for bonus points, there's something in [fragment_cache](https://github.com/rails-api/active_model_serializers/blob/61412d80cabc3900dfbd886bc3f25fb0e0d39433/lib/active_model_serializers/fragment_cache.rb#L112-L113) that also does some class stringify munging that looks similar to the above, and we might want to include)